### PR TITLE
Implemented changes to positions and group position memberships

### PR DESCRIPTION
**Pull request by majesticfish**
### Summary
- Deleted group_members table
- Changed queries of group_members table to quires of position_holders NATURAL JOIN positions
   - This adheres to the paradigm that all members of groups have an associated position
- Created a position relation table
### Test Plan
-

### CODE REVIEW

File: sql/donut.sql [hunk 1]
```diff
@@ -173,32 +179,11 @@ CREATE VIEW group_house_membership AS (
...
-                                       -- can send emails to group
-    control     BOOLEAN DEFAULT FALSE, -- Toggles whether or not this position
-                                       -- has admin control over group
-    receive     BOOLEAN DEFAULT TRUE,  -- Toggles if this position receives
-                                       -- emails from this group
-    PRIMARY KEY (pos_id, group_id),
-    FOREIGN KEY (pos_id) REFERENCES positions(pos_id),
-    FOREIGN KEY (group_id) REFERENCES groups(group_id)
-);
+CREATE TABLE position_relations (
```
Comment (RobertEng) [inline sql/donut.sql]: Are there any tests for this table?

File: donut/modules/core/helpers.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def get_group_list_of_member(user_id):
         result: All the groups that an user_id is a part of
     """
     query = """SELECT group_id, group_name, control
-    FROM group_members NATURAL JOIN groups NATURAL JOIN members
+    FROM positions NATURAL JOIN groups NATURAL JOIN position_holders
```
Comment (calebsander) [inline donut/modules/core/helpers.py]: Doesn't this need to take `positions_relations` into account?

File: donut/modules/groups/helpers.py [hunk 3]
```diff
@@ -138,7 +138,7 @@ def get_position_data(fields=None):
 
 
 def get_members_by_group(group_id):
-    query = "SELECT user_id FROM group_members "
+    query = "SELECT user_id FROM positions NATURAL JOIN position_holders "
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Doesn't this need to take `positions_relations` into account?

File: donut/modules/core/helpers.py [hunk 4]
```diff
@@ -129,7 +129,7 @@ def get_group_list_of_member(user_id):
         result: All the groups that an user_id is a part of
     """
```
Comment (RobertEng) [inline donut/modules/core/helpers.py:130]: Should this function be called get_member_groups

File: sql/donut.sql [hunk 5]
```diff
@@ -173,32 +179,11 @@ CREATE VIEW group_house_membership AS (
...
+    relation_id   INT  NOT NULL AUTO_INCREMENT,
```
Comment (calebsander) [inline sql/donut.sql]: Could just use `(pos_id_from, pos_id_to)` as your PK

**Review by calebsander — CHANGES_REQUESTED** [review]
Generally looks good, although I think you aren't making use of `positions_relations`. Do the tests all still work? Also, I think we should remove the `group_id` column from `position_holders` since it should be specified by the referenced position.

File: sql/donut.sql [hunk 6]
```diff
@@ -173,32 +179,11 @@ CREATE VIEW group_house_membership AS (
...
+    pos_id_from   INT  NOT NULL,
+    pos_id_to     INT  NOT NULL,
```
Comment (RobertEng) [inline sql/donut.sql]: It looks like this table should be used in some helper functions but is not. Add a comment for what this table is.

File: sql/donut.sql [hunk 7]
```diff
@@ -139,6 +138,13 @@ CREATE TABLE positions (
     group_id INT         NOT NULL,
     pos_id   INT         NOT NULL AUTO_INCREMENT,
     pos_name VARCHAR(32) NOT NULL,
+    send        BOOLEAN DEFAULT FALSE, -- Toggles whether or not this position
```
Comment (RobertEng) [inline sql/donut.sql:141]: send, control, and receive have their defaults defined here, but also need to be changed on a person by person basis.

**Review by RobertEng — CHANGES_REQUESTED** [review]

File: sql/donut.sql (hunk 7 above)
Comment (majesticfish) [reply sql/donut.sql:141]: Sorry Whats the problem with that? The default is just the value that is set when theres nothing there right?

File: sql/donut.sql (hunk 1 above)
Comment (majesticfish) [reply sql/donut.sql]: no I'll write them

File: donut/modules/core/helpers.py (hunk 2 above)
Comment (majesticfish) [reply donut/modules/core/helpers.py]: yup

**Review by joshc — APPROVED** [review]
lgtm - further work needed on newsgroups part

File: sql/test_data.sql [hunk 8]
```diff
@@ -47,6 +47,9 @@ INSERT INTO members(
     1,
     '203'
 );
+INSERT INTO members(uid, last_name, first_name, email, phone) VALUES
+    ('2045251', 'Yu', 'Sean', 'ssyu@caltech.edu', NULL);
```
Comment (calebsander) [inline sql/test_data.sql:51]: `phone` doesn't have to be specified, but minor point

**Review by calebsander — APPROVED** [review]

**Pull request merged**